modules/schematics/src/create-action-migration/index.ts

```diff
@@ -0,0 +1,206 @@
+            return undefined;
+          }
+
+          const payloadType = getConstructorPayloadType(
+            host,
+            path,
+            actionClass
+          );
+          if (payloadType) {
+            importProps = true;
+          }
+
+          const createActionDefinition = composeCreateAction(
+            typeValue,
+            payloadType,
+            actionClass
+          );
+
+          if (createActionDefinition.includes('props')) {
+            importProps = true;
+          }
+
+          return [
+            new RemoveChange(
+              sourceFile.fileName,
+              typeVariable[0].getStart(sourceFile),
+              typeVariable[0].getEnd()
+            ),
+            new RemoveChange(
+              sourceFile.fileName,
+              actionClass.getStart(sourceFile),
+              actionClass.getEnd()
+            ),
+            new InsertChange(
+              path,
+              typeVariable[0].getStart(sourceFile),
+              createActionDefinition
+            ),
+          ];
+        })
+        .filter(Boolean) as []).reduce(
+        (imports, curr) => imports.concat(curr),
+        []
+      );
+
+      let importChanges: (ReplaceChange | RemoveChange | Change)[] = [];
+      const imports = [];
+      if (changes.length > 0) {
+        imports.push('createAction');
+        if (importProps) {
+          imports.push('props');
+        }
+        importChanges = replaceImport(
+          sourceFile,
+          path,
+          '@ngrx/store',
+          'Action',
+          imports.join(', ')
+        );
+      }
+
+      return commitChanges(host, sourceFile.fileName, [
+        ...changes,
+        ...importChanges,
+      ]);
+    });
+}
+
+const isActionImported = (sourceFile: ts.SourceFile): boolean => {
+  return !!sourceFile.statements
+    .filter(ts.isImportDeclaration)
+    .filter(({ moduleSpecifier }) =>
+      moduleSpecifier.getText(sourceFile).includes('@ngrx/store')
+    )
+    .filter(p => {
+      return (p.importClause!
+        .namedBindings! as ts.NamedImports).elements.filter(spec => {
+        if (spec.name.text) {
+          return spec.name.text === 'Action';
+        }
+      }).length;
+    }).length;
+};
+
+const getActionClasses = (sourceFile: ts.SourceFile): ts.ClassDeclaration[] => {
+  return sourceFile.statements
+    .filter(ts.isClassDeclaration)
+    .filter(
+      statement => statement.heritageClauses && statement.heritageClauses.length
+    )
+    .filter(statement => {
+      return statement.heritageClauses!.filter(clause => {
+        return (clause.types || []).filter(type => {
+          return (
+            type.expression && type.expression.getText(sourceFile) === 'Action'
+          );
+        }).length;
+      }).length;
+    });
+};
+
+const getConstructorPayloadType = (
+  host: Tree,
+  path: Path,
+  actionClass: ts.ClassDeclaration
+): string => {
+  let payloadType = '';
+
+  let payloadInConstructor = actionClass.members
```

Comment:
this would mean we would only convert one property, right?
e.g. below, we would only use `foo`?

```
constructor(foo: string, bar: string){}
```